hacking/contrib.md

```diff
@@ -21,4 +21,13 @@
 without further testing assuming they conform to style and don't do anything stupid.  The author becomes the maintainer for that target.
  - If the code is orphaned and no maintainer can be found in reasonable time, the patch author may take over as the new maintainer.
  - The current maintainer should be noted in the copyright banner at the top of the file in question.
- - The Github request review feature of pull requests should be used to request a review from the maintainer.
+ - The GitHub request review feature of pull requests should be used to request a review from the maintainer.
+
+## Checks before providing pull requests
+
+Before providing a pull request, please:
+- Check and remove trailing whitespace: `git rebase --whitespace=fix ...`
```

Comment:
This should be done if you're using `clang-format` every commit. It might be better to tell people to install and use `clang-format` instead as that's what our linters are expecting.

Reply:
In line with my comment above: how about adding instructions to install `clang-format` and installing the `pre-commit` in the [Contribution Guidelines](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to keep everything in once place?

Reply:
Agreed